### PR TITLE
Relax restrictions on nested groups

### PR DESCRIPTION
**Pull request by tranma**
I don't remember why it was restricted like this before (we support nested groups now). Some existing dictionaries won't compile with these restrictions.

! @amosr

### CODE REVIEW

File: icicle-source/src/Icicle/Source/Checker/Invariants.hs [hunk 1]
```diff
@@ -48,16 +48,13 @@ invariantQ ctx (Query (c:cs) xfinal)
...
 
     Distinct _ x
-     -> goX x >> goBanWindowAndGroupFold
+     -> goX x >> go
 
     GroupFold _ _ _ x
-     | allowWindows inv
-     -> goX x >> goBanWindowAndGroupFold
-     | otherwise
-     -> errBanGroupFold
```
Comment (tranma) [inline icicle-source/src/Icicle/Source/Checker/Invariants.hs]: @amosr I think it should be stricter than this, but what we had before was too strict?

Comment (amosr) [inline icicle-source/src/Icicle/Source/Checker/Invariants.hs]: something about - we need to ban latests inside groups? hmm

Comment (amosr) [inline icicle-source/src/Icicle/Source/Checker/Invariants.hs]: we might need to dig up the change that made it stricter to see why

Comment (tranma) [inline icicle-source/src/Icicle/Source/Checker/Invariants.hs]: the change actually made it less strict already, from the group in group change: https://github.com/ambiata/icicle/commit/88901cd08b51f4988913c000142ee8f74e5f616e#diff-743f2c2eb9de71059af5941e74e658c6

File: icicle-source/src/Icicle/Source/Checker/Invariants.hs [hunk 2]
```diff
@@ -48,16 +48,13 @@ invariantQ ctx (Query (c:cs) xfinal)
      -> errBanLatest
 
     GroupBy _ x
-     -> goX x >> goBanWindowAndGroupFold
+     -> goX x >> go
```
Comment (amosr) [inline icicle-source/src/Icicle/Source/Checker/Invariants.hs]: I think windows still can't be inside groups or distincts

File: icicle-source/src/Icicle/Source/Checker/Invariants.hs (hunk 1 above)
Comment (amosr) [inline icicle-source/src/Icicle/Source/Checker/Invariants.hs]: ah. so I think we need to ban latests here, #468

Comment (amosr) [inline icicle-source/src/Icicle/Source/Checker/Invariants.hs]: if we ban latests I think we can reenable the test #488

**Comment by amosr** [comment]
is this superseded by #513 ?

**Comment by tranma** [comment]
yeah this is really the segfault-debugging PR

**Pull request closed without merging**